File: src/test/scala/dregex/EmbeddedFlagTest.scala
package dregex

import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.function.Executable

class EmbeddedFlagTest {

  @Test
  def testEmbeddedFlags() = {
    // OK
    Regex.compile("(?x)a")

    // flags in the middle
    assertThrows(classOf[InvalidRegexException], new Executable {
      def execute() = Regex.compile(" (?x)a")
    })
    assertThrows(classOf[InvalidRegexException], new Executable {
      def execute() = Regex.compile("(?x)a(?x)")
    })

    // unknown flag
    assertThrows(classOf[InvalidRegexException], new Executable {
      def execute() = Regex.compile("(?w)a")
    })
  }

}

File: src/test/java/dregex/EmbeddedFlagTest.java
package dregex;

import static org.junit.jupiter.api.Assertions.assertThrows;

import org.junit.jupiter.api.Test;

class EmbeddedFlagTest {

    @Test
    void testEmbeddedFlags() {

        // OK
        Regex.compile("(?x)a");

        // flags in the middle
        assertThrows(InvalidRegexException.class, () -> Regex.compile(" (?x)a"));
        assertThrows(InvalidRegexException.class, () -> Regex.compile("(?x)a(?x)"));

        // unknown flag
        assertThrows(InvalidRegexException.class, () -> Regex.compile("(?w)a"));
    }
}
